hail/python/hailtop/hailctl/batch/cli.py

```diff
@@ -164,6 +164,9 @@ def submit(
     files: Ann[
         Optional[List[str]], Opt(help='Files or directories to add to the working directory of the job.')
     ] = None,
+    mounts: Ann[
+        Optional[List[str]], Opt(help='Files or directories to add to a specific mount location in the job. Equivalent to -v src:dest in Docker.')
+    ] = None,
```

Comment:
I don't like the term `mount` b/c this isn't mounting the users hard drive into the cloud.

I think it's a bit confusing that there are two arguments with nearly identical descriptions.

How about this:
1. Replace `--file` with the functionality provided by `--mount`
2. Allow the new `--file` to elide the destination. When the destination is elided, it is assumed to be the same as the source (so: `..` means `..:..` both relative to the current working directory)
3. By default, make the working directory of the job match the working directory of the user's laptop. 

These three things together allow `--file .. --file ../../reference_data:/reference_data --file ../../other_project/thing.py` to work as expected. You'll get a container with this file structure:
```
/Users
  /dking
    /projects
      /foo
        thing.py
        ... everything else in the working directory of this script on the user's laptop ...
/reference_data
   ... everything in /Users/dking/projects/reference_data on the user's laptop ...
```

NB: if you do `--file .. --file foo.py` you're implicitly copying foo.py twice. Unnecessary, but should be fine if foo.py is not mutated while we are submitting.